lib/rails_state_machine/event.rb

```diff
@@ -64,8 +71,23 @@ def allowed_from?(state_name)
       @transitions_by_state_name.key?(state_name&.to_sym)
     end
 
-    def future_state_name(state_name)
+    def future_state_name(state_name, raise_for_invalid_transitions)
```

Comment:
This is a public method of the API which is being called in some production projects, so this is a breaking change.
Also, adding to `@volatile_before_validation` is a bit weird when this method is called to retrieve the target state name but adds validation logic on the instance.

It probably did not make sense in the first place that calling this method would raise an error.

Instead, it should return either the target state name or `nil`, and places calling `future_state_name` need to raise an error if they were not expecting `nil`.

I'm aware that my suggestion is also a breaking API change IMO it would be a useful an improvement.